Packs/Okta/Integrations/OktaLog/OktaLog.py

```diff
@@ -0,0 +1,196 @@
+            return response
+        except Exception as exc:
+            msg = f'something went wrong with the http call {exc}'
+            LOG(msg)
+            raise DemistoException(msg) from exc
+
+    def set_next_run_filter(self, after: str):
+        self.request.params.set_since_value(after)  # type: ignore
+
+
+class GetEvents:
+    """
+    A class to handle the flow of the integration
+    """
+
+    def __init__(self, client: Client) -> None:
+        self.client = client
+
+    def make_api_call(self):
+        limit_tmp = int(self.client.request.params.limit)  # type: ignore
+        if limit_tmp > 1000:
+            self.client.request.params.limit = '1000'  # type: ignore
+        response = self.client.call()
+        events: list = response.json()
+        self.client.request.params.limit = str(limit_tmp - len(events))  # type: ignore
+        return events
+
+    def _iter_events(self, last_object_ids: list) -> None:  # type: ignore
+        """
+        Function that responsible for the iteration over the events returned from the Okta api
+        """
+
+        events: list = self.make_api_call()  # type: ignore
+        if last_object_ids:
+            events = GetEvents.remove_duplicates(events, last_object_ids)
+        while True:
+            yield events
+            last = events[-1]
+            self.client.set_next_run_filter(last['published'])
+            events: list = self.make_api_call()  # type: ignore
+            try:
+                assert events
+            except (IndexError, AssertionError):
+                LOG('empty list, breaking')
+                break
+
+    def aggregated_results(self, last_object_ids: List[str] = None) -> List[dict]:
+        """
+        Function to group the events returned from the api
+        """
+        stored_events = []
+        for events in self._iter_events(last_object_ids):  # type: ignore
+            stored_events.extend(events)
+            if int(self.client.request.params.limit) == 0 or len(events) == 0:  # type: ignore
+                break
+        return stored_events
+
+    @staticmethod
+    def get_last_run(events: List[dict]) -> dict:
+        """
+        Get the info from the last run, it returns the time to query from and a list of ids to prevent duplications
+        """
+
+        ids = []
+        # gets the last event time
+        last_time = events[-1].get('published')
+        for event in reversed(events):
+            if event.get('published') != last_time:
+                break
+            ids.append(event.get('uuid'))
+        last_time = datetime.strptime(str(last_time).lower().replace('z', ''), '%Y-%m-%dt%H:%M:%S.%f')
+        return {'after': last_time.isoformat(), 'ids': ids}
+
+    @staticmethod
+    def remove_duplicates(events: list, ids: list) -> list:
+        """
+        Remove object duplicates by the uuid of the object
+        """
+        return [event for event in events if event['uuid'] not in ids]
+
+
+def main():
```

Comment:
Add a bare try-except around the logic of the main as we have for all integrations